src/infil.c

```diff
@@ -233,6 +257,7 @@ void infil_getState(int j, int m, double x[])
 //  Purpose: retrieves the current infiltration state for a subcatchment.
 //
 {
+    m = Subcatch[j].infil_type; // 2016-01-15: CHANGE PASS-BY-VALUE, m
```

Comment:
@aerispaha, `m` is an input argument so I think the caller of `infil_getState` should continue to pass in the infiltration type instead of setting it here.